agent/test-agent-cli/src/get_secret.rs

```diff
@@ -0,0 +1,49 @@
+use crate::error::{
+    ClientSnafu, ConfMissingSnafu, ConversionSnafu, Result, SecretKeyFetchSnafu, SecretMissingSnafu,
+};
+use crate::init::TestConfig;
+use agent_common::secrets::SecretsReader;
+use argh::FromArgs;
+use snafu::{OptionExt, ResultExt};
+use test_agent::{Client, DefaultClient, Spec};
+
+#[derive(Debug, FromArgs, PartialEq)]
+#[argh(subcommand, name = "get-secret", description = "Get secret for a key")]
```

Comment:
It may be easier of the user if either this is based on the secret name not the key or init doesn't return the secrets map.